analysis/exponential.lean

```diff
@@ -397,6 +453,72 @@ lemma exists_sin_eq {x : ℝ} (hx₁ : -1 ≤ x) (hx₂ : x ≤ 1) : ∃ y, -(π
   (by rwa [sin_neg, sin_pi_div_two]) (by rwa sin_pi_div_two)
   (le_trans (neg_nonpos.2 (le_of_lt pi_div_two_pos)) (le_of_lt pi_div_two_pos))
 
+/-- The type of angles -/
+def angle : Type :=
+quotient_add_group.quotient (gmultiples (2 * π))
+
+instance angle.add_comm_group : add_comm_group angle :=
+quotient_add_group.add_comm_group _
+
+instance angle.has_coe : has_coe ℝ angle :=
+⟨quotient.mk'⟩
+
+instance angle.is_add_group_hom : is_add_group_hom (coe : ℝ → angle) :=
+@quotient_add_group.is_add_group_hom _ _ _ (normal_add_subgroup_of_add_comm_group _)
+
+@[simp] lemma coe_zero : ↑(0 : ℝ) = (0 : angle) := rfl
+@[simp] lemma coe_add (x y : ℝ) : ↑(x + y : ℝ) = (↑x + ↑y : angle) := rfl
+@[simp] lemma coe_neg (x : ℝ) : ↑(-x : ℝ) = -(↑x : angle) := rfl
+@[simp] lemma coe_sub (x y : ℝ) : ↑(x - y : ℝ) = (↑x - ↑y : angle) := rfl
+@[simp] lemma coe_gsmul (x : ℝ) (n : ℤ) : ↑(gsmul n x : ℝ) = gsmul n (↑x : angle) :=

```

Comment:
Hmm, does this not follow by some lemma from `angle.is_add_group_hom`?

Reply:
yes `is_add_group_hom.gsmul`. Also, these lemmas should be in the `angle` namespace, which itself should be in the `real` namespace, which I think it already is.